pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -170,3 +172,17 @@ func (p *InstanceTypeProvider) isDefaultInstanceType(instanceTypeInfo *packing.I
 		!*instanceTypeInfo.BareMetal &&
 		functional.HasAnyPrefix(*instanceTypeInfo.InstanceType, "m", "c", "r", "a", "t3", "t4")
 }
+
+func (p *InstanceTypeProvider) isArchitectureSupported(architecture *string, instance *packing.Instance) bool {
+	return architecture == nil ||
+		functional.ContainsString(aws.StringValueSlice(instance.ProcessorInfo.SupportedArchitectures), *architecture)
+}
+
+func (p *InstanceTypeProvider) isCapacityTypeSupported(capacityType string, instance *packing.Instance) bool {
+	return capacityType == "" ||
```

Comment:
aren't we guaranteed that capacityType has been defaulted?

Reply:
same answer as above.